spec/client_fork_spec.rb

```diff
@@ -0,0 +1,169 @@
+# Copyright 2016-2018 The NATS Authors
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+return unless Process.respond_to?(:fork) # Skip if fork is not supported (Windows, JRuby, etc)
+return unless Process.respond_to?(:_fork) # Skip if around fork callbacks are not supported (before Ruby 3.1)
+
+require 'spec_helper'
+
+describe 'Client - Fork detection' do
+
+  before(:all) do
+    @tmpdir = Dir.mktmpdir("ruby-jetstream-fork")
+    @s = NatsServerControl.new("nats://127.0.0.1:4524", "/tmp/test-nats.pid", "-js -sd=#{@tmpdir}")
+    @s.start_server(true)
+  end
+
+  after(:all) do
+    @s.kill_server
+    FileUtils.remove_entry(@tmpdir)
+  end
+
+  class Component
```

Comment:
It's better to define temporary constants via `stub_const` or add a cleanup code to remove them manually; otherwise, the `Component` leaks to the global state and can collide with other similar constants in the future.

Reply:
Seems that there is no need in this auxiliary test altogether. Removed it :scissors: